qiskit/chemistry/algorithms/ground_state_solvers/orbital_optimization_vqe.py

```diff
@@ -0,0 +1,663 @@
+    amplitudes of the ground state of given operator such that in the end a given ansatz
+    can be closest to that ground state, producing larger overlap and lower eigenvalue than
+    conventional VQE.  Kappa is parametrized and optimized inside the OOVQE in the same way as
+    the gate angles. Therefore, at each step of OOVQE the coefficient matrix C is modified and
+    the operator is recomputed, unlike usual VQE where operator remains constant.
+    Iterative OO refers to optimization in two steps, first the wavefunction and then the
+    orbitals. It allows for faster optimization as the operator is not recomputed when
+    wavefunction is optimized. It is recommended to use the iterative method on real device/qasm
+    simulator with noise to facilitate the convergence of the classical optimizer.
+    For more details of this method refer to: https://aip.scitation.org/doi/10.1063/1.5141835
+    """
+
+    def __init__(self,
+                 transformation: FermionicTransformation,
+                 solver_factory: MinimumEigensolverFactory,
+                 initial_point: Optional[np.ndarray] = None,
+                 orbital_rotation: Optional['OrbitalRotation'] = None,
+                 bounds: Optional[np.ndarray] = None,
+                 iterative_oo: bool = True,
+                 iterative_oo_iterations: int = 2,
+                 ):
+        """
+        Args:
+            transformation: a fermionic driver to operator transformation strategy.
+            solver_factory: a factory for the VQE solver employing any custom variational form.
+            initial_point: An optional initial point (i.e. initial parameter values)
+                for the optimizer. If ``None`` then VQE will look to the variational form for a
+                preferred point and if not will simply compute a random one.
+            orbital_rotation: instance of
+                :class:`~qiskit.chemistry.ground_state_calculation.OrbitalRotation` class
+                that creates the matrices that rotate the orbitals needed to produce the rotated
+                MO coefficients C as C = C0 * exp(-kappa).
+            bounds: bounds for variational form and orbital rotation
+                 parameters given to a classical optimizer.
+            iterative_oo: when ``True`` optimize first the variational form and then the
+                orbitals, iteratively. Otherwise, the wavefunction ansatz and orbitals are
+                optimized simultaneously.
+            iterative_oo_iterations: number of iterations in the iterative procedure,
+                set larger to be sure to converge to the global minimum.
+        Raises:
+            AquaError: if the number of orbital optimization iterations is less or equal to zero.
+        """
+
+        super().__init__(transformation, solver_factory)
+        # todo: no need, initializer in the super class
+        self._solver_factory = solver_factory
```

Comment:
Yes the solver should be accessed from and stored just in the base class. Please change the code to use the solver getter from there to access the solver and remove this

Reply:
done.